### PR TITLE
[libSwiftScan] Support response file as input arguments

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Build system can pass response file to libSwiftScan as the command-line arguments. libSwiftScan needs to expand the response file before  performing tasks.

rdar://121291342

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

File: tools/libSwiftScan/SwiftCaching.cpp [hunk 1]
```diff
@@ -196,6 +197,25 @@ swiftscan_string_ref_t swiftscan_cas_store(swiftscan_cas_t cas, uint8_t *data,
       CAS.getID(*Result).toString().c_str());
 }
 
+// Expand the invocation if there is repsonseFile into Args that are passed in
```
Comment (benlangmuir) [inline tools/libSwiftScan/SwiftCaching.cpp:200]: Should be a `///` comment

File: tools/libSwiftScan/SwiftCaching.cpp [hunk 2]
```diff
@@ -270,11 +286,12 @@ computeCacheKeyFromIndex(llvm::cas::ObjectStore &CAS,
 swiftscan_string_ref_t
 swiftscan_cache_compute_key(swiftscan_cas_t cas, int argc, const char **argv,
                             const char *input, swiftscan_string_ref_t *error) {
-  std::vector<const char *> Compilation;
-  for (int i = 0; i < argc; ++i)
-    Compilation.push_back(argv[i]);
+  llvm::SmallVector<const char *> Compilation;
```
Comment (benlangmuir) [inline tools/libSwiftScan/SwiftCaching.cpp:289]: Nit: I would rename this "ArgsStorage" or similar to avoid accidentally using this.  Same below.

File: test/CAS/binary_module_deps.swift [hunk 3]
```diff
@@ -10,7 +10,7 @@
 // RUN: %validate-json %t/deps.json | %FileCheck %s -DTEMP=%t
 
 /// Test binary module key: binary module key is the CASID of itself.
-// RUN: %S/Inputs/SwiftDepsExtractor.py %t/deps.json swiftPrebuiltExternal:A moduleCacheKey > %t/A.key.casid
+// RUN: %{python} %S/Inputs/SwiftDepsExtractor.py %t/deps.json swiftPrebuiltExternal:A moduleCacheKey > %t/A.key.casid
```
Comment (benlangmuir) [inline test/CAS/binary_module_deps.swift:13]: Are these changes related?

Comment (cachemeifyoucan) [reply test/CAS/binary_module_deps.swift:13]: Let me put this into a separate PR.

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test windows platform

**Review by artemcm — APPROVED** [review]
Thank you.
